dice/rollvm_migrate.go

```diff
@@ -0,0 +1,567 @@
+			Attrs: m,
+		}
+		return ds.NewComputedValRaw(cd)
+	default:
+		return ds.NewNullVal()
+	}
+}
+
+func dsValueToRollVMv1(v *ds.VMValue) *VMValue {
+	var v2 *VMValue
+	switch v.TypeId {
+	case ds.VMTypeInt:
+		v2 = &VMValue{TypeID: VMTypeInt64, Value: int64(v.MustReadInt())}
+	case ds.VMTypeFloat:
+		v2 = &VMValue{TypeID: VMTypeInt64, Value: int64(v.MustReadFloat())}
+	default:
+		v2 = &VMValue{TypeID: VMTypeString, Value: v.ToString()}
+	}
+	return v2
+}
+
+func DiceFormatTmplV1(ctx *MsgContext, s string) string { //nolint:revive
+	var text string
+	a := ctx.Dice.TextMap[s]
+	if a == nil {
+		text = "<%未知项-" + s + "%>"
+	} else {
+		text = ctx.Dice.TextMap[s].Pick().(string)
+	}
+	return DiceFormat(ctx, text)
+}
+
+func DiceFormatV1(ctx *MsgContext, s string) string { //nolint:revive
+	s = CompatibleReplace(ctx, s)
+
+	r, _, _ := ctx.Dice._ExprTextV1(s, ctx)
+	return r
+}
+
+func DiceFormat(ctx *MsgContext, s string) string {
+	ret, err := DiceFormatV2(ctx, s)
+	if err != nil {
+		// 遇到异常，尝试一下V1
+		return DiceFormatV1(ctx, s)
+	}
+	return ret
+}
+
+func DiceFormatTmpl(ctx *MsgContext, s string) string {
+	ret, err := DiceFormatTmplV2(ctx, s)
+	if err != nil {
+		// 遇到异常，尝试一下V1
+		return DiceFormatTmplV1(ctx, s)
+	}
+	return ret
+}
+
+func (ctx *MsgContext) Eval(expr string, flags *ds.RollConfig) *VMResultV2 {
+	ctx.CreateVmIfNotExists()
+	vm := ctx.vm
+	if flags != nil {
+		vm.Config = *flags
+	}
+	err := vm.Run(expr)
+
+	if err != nil {
+		return &VMResultV2{vm: vm}
+	}
+	return &VMResultV2{VMValue: *vm.Ret, vm: vm}
+}
+
+// EvalFString TODO: 这个名字得换一个
+func (ctx *MsgContext) EvalFString(expr string, flags *ds.RollConfig) *VMResultV2 {
+	expr = CompatibleReplace(ctx, expr)
+
+	// 隐藏的内置字符串符号 \x1e
+	r := ctx.Eval("\x1e"+expr+"\x1e", flags)
+	if r.vm.Error != nil {
+		fmt.Println("脚本执行出错: ", expr, "->", r.vm.Error)
+	}
+	return r
+}
+
+type VMResultV2m struct {
+	*ds.VMValue
+	vm        *ds.Context
+	legacy    *VMResult
+	cocPrefix string
+}
+
+func (r *VMResultV2m) GetAsmText() string {
+	if r.legacy != nil {
+		return r.legacy.Parser.GetAsmText()
+	}
+	return r.vm.GetAsmText()
+}
+
+func (r *VMResultV2m) IsCalculated() bool {
+	if r.legacy != nil {
+		return r.legacy.Parser.Calculated
+	}
+	return r.vm.IsCalculateExists()
+}
+
+func (r *VMResultV2m) GetRestInput() string {
+	if r.legacy != nil {
+		return r.legacy.restInput
+	}
+	return r.vm.RestInput
+}
+
+func (r *VMResultV2m) GetMatched() string {
+	if r.legacy != nil {
+		return r.legacy.Matched
+	}
+	return r.vm.Matched
+}
+
+func (r *VMResultV2m) GetCocPrefix() string {
+	if r.legacy != nil {
+		return r.legacy.Parser.CocFlagVarPrefix
+	}
+	return r.cocPrefix
+}
+
+func (r *VMResultV2m) GetVersion() int64 {
+	if r.legacy != nil {
+		return 1
+	}
+	return 2
+}
+
+func (r *VMResultV2m) ToString() string {
+	if r.legacy != nil {
+		return r.legacy.ToString()
+	}
+	return r.VMValue.ToString()
+}
+
+// DiceExprEvalBase 向下兼容执行，首先尝试使用V2执行表达式，如果V2失败，fallback到V1
+// Deprecated: 不建议用，纯兼容旧版
```

Comment:
更符合 go 注释格式规范的写法，其余各处类似

```suggestion
// DiceExprEvalBase 向下兼容执行，首先尝试使用V2执行表达式，如果V2失败，fallback到V1
//
// Deprecated: 不建议用，纯兼容旧版
```